activerecord/lib/active_record/migration.rb

```diff
@@ -967,7 +967,7 @@ def schema_migrations_table_name
       end
 
       def get_all_versions(connection = Base.connection)
-        if connection.table_exists?(schema_migrations_table_name)
+        if connection.data_source_exists?(schema_migrations_table_name)
```

Comment:
I agree, `SchemaMigration.create_table` uses `connection.create_table` internally, so I will change back it to `table_exists?`.

```ruby
      def create_table(limit=nil)
        unless table_exists?
          version_options = {null: false}
          version_options[:limit] = limit if limit

          connection.create_table(table_name, id: false) do |t|
            t.column :version, :string, version_options
          end
          connection.add_index table_name, :version, unique: true, name: index_name
        end
      end
```